publish-extensions.js

```diff
@@ -14,11 +14,24 @@ const ovsx = require('ovsx');
 const path = require('path');
 const util = require('util');
 const semver = require('semver');
+const download = require('download');
 const exec = require('./lib/exec');
 const readFile = util.promisify(fs.readFile);
 
 (async () => {
-  /** @type {{ extensions: { id: string, repository: string, version?: string, checkout?: string, location?: string, prepublish?: string }[] }} */
+  /**
+   * @type {{
+   *    extensions: {
+   *        id: string,
+   *        repository: string,
+   *        version?: string,
+   *        checkout?: string,
+   *        location?: string,
+   *        prepublish?: string,
+   *        download?: string
+   *    }[]
```

Comment:
Maybe separate the two types to be more precise? (E.g. `repository` is mandatory in one case, but should be omitted in the other).

```suggestion
   *    extensions: {
   *        id: string,
   *        repository: string,
   *        version?: string,
   *        checkout?: string,
   *        location?: string,
   *        prepublish?: string
   *    }[] | {
   *        id: string,
   *        download: string,
   *        version?: string
   *    }[]
```

Reply:
That's problematic, ts-check then complains when trying to access properties that are only in one of the two alternatives.